src/js/fileupload2.js

```diff
@@ -663,14 +667,28 @@ function queueFileForDirectUpload(file) {
     }
     var fUpload = new fileUpload(file);
     let send = true;
-    let path = file.webkitRelativePath.substring(file.webkitRelativePath.indexOf('/') + 1);
-    console.log(path);
+    let origPath = file.webkitRelativePath.substring(file.webkitRelativePath.indexOf('/') + 1);
+    
+    //Remove filename part
+    let path =origPath.substring(0, origPath.length - file.name.length);
+    let badPath = (path.match(/^[\w\d_\-\.\\\/ ]*$/)===null);
+    if(badPath) {
+      if($('.warn').length==0) {
+        addMessage('warn', 'msgRequiredPathOrFileNameChange');
+      }
+      //Munge path according to rules
+      path = path.replace(/[^\w\d_\\.\\\/ ]+/g,'_');

```

Comment:
A typo - with simplification from above this would be `/[^\w\-\.\\\/ ]+/g`  (- was missing)